pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -331,6 +343,53 @@ func (c *EtcdCertSignerController) createSecretForNode(node *corev1.Node) error
 	return nil
 }
 
+func createPeerDNSSubjectAlternativeNamesFromMemberList(etcdDiscoveryDomain string, etcdMembers []*etcdserverpb.Member, nodeInternalIPs []string) []string {
+	var found bool
+	var errs []string
+	hosts := []string{"localhost"}
+	for _, m := range etcdMembers {
+		// 4.3 naming
+		if strings.HasPrefix(m.Name, "etcd-member") {
+			found = true
+		}
+		if len(m.PeerURLs) > 0 && len(m.ClientURLs) > 0 {
+			for _, p := range m.PeerURLs {
+				peer, err := url.Parse(p)
+				if err != nil {
+					errs = append(errs, err.Error())
+					continue
+				}
+				if addr := net.ParseIP(peer.Hostname()); addr != nil {
+					continue
+				}
+				for _, clientUrl := range m.ClientURLs {
+					client, err := url.Parse(clientUrl)
+					if err != nil {
+						errs = append(errs, err.Error())
+						continue
+					}
+					for _, nodeIP := range nodeInternalIPs {
+						if nodeIP == client.Hostname() {
+							hosts = append(hosts, peer.Hostname())
+						}
+					}
+				}
+			}
+		}
+	}
+
+	switch {
+	case len(errs) > 0:
+		break // fallback to wildcard
```

Comment:
Is it possible to have no errors but the len(hosts) is zero?

Reply:
Never mind. Then, also it will fall back to wild-card.

Reply:
yeah and it won't ever be 0 

https://github.com/openshift/cluster-etcd-operator/pull/340/files/c1734e4befc41128038c407ac72d611a8f309347#diff-97a182619797a0137f8ca3c8b467bae4R349